src/wp-includes/theme.php

```diff
@@ -836,6 +836,15 @@ function switch_theme( $stylesheet ) {
 
 	update_option( 'theme_switched', $old_theme->get_stylesheet() );
 
+	/*
+	 * Reset globals to force refresh the next time these directories are accessed 
+	 * via `locate_template()` as long as we're not in a switched context.
+	 */
+	if ( isset( $GLOBALS['switched'] ) && ! $GLOBALS['switched'] ) {
```

Comment:
Better to use the dedicated function:
```suggestion
	if ( ! is_multisite() || ! ms_is_switched() ) {
```

Though per my above comment, we may want to not reset the globals to `null` here, but actually refresh their value right away, since we don't know from which context `locate_template()` will be called next.

Reply:
Implemented in [8341dbc](https://github.com/WordPress/wordpress-develop/pull/5926/commits/8341dbc4f0a021ce6170c8f91ee1ec796725cf69)